Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeNode.mm

```diff
@@ -248,9 +248,30 @@
 }
 
 #if ENABLE(THREADED_ANIMATION_RESOLUTION)
-void RemoteLayerTreeNode::setAcceleratedEffectsAndBaseValues(const WebCore::AcceleratedEffects& effects, const WebCore::AcceleratedEffectValues& baseValues)
+void RemoteLayerTreeNode::setAcceleratedEffectsAndBaseValues(const WebCore::AcceleratedEffects& effects, const WebCore::AcceleratedEffectValues& baseValues, RemoteLayerTreeHost* host)
 {
     ASSERT(isUIThread());
+    ASSERT(host);
+
+    if (m_effects) {
+        m_effects->clear(layer());
+        host->animationEffectStackRemoved(*m_effects);
+    }
```

Comment:
Do we need to do this if `effects` is not empty?

Reply:
I *think* we do, though I'm not sure super about how bad it would be to not. The comment below tries to explain the case, but I can go into a bit more detail.

Imagine we had an accelerated animation running, and it's part way through.

The web page then makes a change replacing it with a new (entirely different) animation, and also adds some other animations (maybe to the same element, maybe to others).

When we receive these animations in the UI process main-thread, we could set the new animation on the existing `RemoteAcceleratedEffectStack` (and create a new `RemoteAcceleratedEffectStack` for other elements that newly have an animation).

We then have a race, where the elements with a new animations don't start displaying anything to do with that animations (including the base value) until the main-thread CA transaction is pushed to CA and deserialised etc. The elements with existing animations could have the new values pushed via the CAPresentationModifier on the animation thread and end up being presented by CA ahead of that.
How bad do we think it is for the set of animation changes (which could be visually dramatic) to not be applied atomically?